yunwoo/programmers-1/카드-뭉치.js

```diff
@@ -0,0 +1,22 @@
+function solution(cards1, cards2, goal) {
+  let result = 'Yes';
+
+  for (const el of goal) {
+    if (cards1[0] === el) {
+      cards1.shift();
+    } else if (cards2[0] === el) {
+      cards2.shift();
+    } else {
+      return (result = 'No');
+    }
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(cards1, cards2, goal) {
  for (const el of goal) {
    if (cards1[0] === el) {
      cards1.shift();
    } else if (cards2[0] === el) {
      cards2.shift();
    } else {
      return 'No';
    }
  }

  return 'Yes';
}
```
- 풀이잘봤습니다! 
- 보니까 result 변수는 필요없어보이네요 바로 문자열을 리턴해도 될 것 같습니다~
- 그리고 shift()를 사용하면 O(N)의 시간복잡도를 갖기때문에 결과적으로 위 풀이는 O(N^2)의 시간복잡도를 갖을 것 같습니다.
- 만약, 효율성을 좀 더 고려한다면 아래와 같이 투 포인터를 이용해서 풀 수 있을 것 같습니다 ~!
```js
function solution(cards1, cards2, goal) {
  let [i, j] = [0, 0];

  for (const word of goal) {
    if (cards1[i] === word) i++;
    if (cards2[j] === word) j++;
  }

  if (i + j === goal.length) return "Yes";
  return "No";
}
```

Reply:
- 와 저번에 사용하신걸 보긴 봤었는데 이렇게 index대신 사용하면 되는군요!!
- 효율과 조건문의 가독성도 마음에 안들었는데 솔루션 감사합니다 👍